### DISCUSSION TITLE
RuntimeException: Session store not set on request.

### DISCUSSION THREAD

**Discussion by mad-coder-365** [livewire/livewire#5674] [1. Help]
I have a Login component

```
class Login extends Component
{
    public string $email;
    public string $password;

    protected $rules = [
        'email' => 'required|email',
        'password' => 'required'
    ];

    public function login(Request $request): RedirectResponse|Redirector|null
    {
        $credentials = $this->validate();

        if (auth()->attempt($credentials)) {
            auth()->user()->update(['last_login_at' => now()]);
            $request->session()->regenerate();
            return redirect()->route('home');
        }

        session()->flash('error', 'Invalid email or password.');

        return null;
    }

    public function render(): View
    {
        return view('livewire.auth.login')->layout('components.layouts.auth');
    }
}
```

I have a unit test for testing login:

```
class LoginTest extends TestCase
{
    use RefreshDatabase, InteractsWithSession;

    function test_can_login(): void
    {
        $this->startSession();

        User::create(['email' => 'test@test.com', 'password' => bcrypt('12345'), 'last_login_at' => now()]);

        Livewire::test('auth.login')

            ->set('email', 'test@test.com')
            ->set('password', '12345')
            ->call('login')
            ->assertRedirect('/home');
    }
}
```

I keep getting this error: ```RuntimeException: Session store not set on request.```

I have tried various answers from the web but have not been able to solve this error. Other tests are working fine. I know this error is due to ```$request->session()->regenerate();``` but I don't know how to make the test pass. 

Please help! Thanks!

**Comment by j3j5** [comment]
I know this is old but I found the same problem while trying to write the tests for my auth components. 

For reference, I'm using Laravel `11.25.0` and Livewire `3.5.8`.

From what I can see, Livewire makes all requests using "test()" function using a class called `RequestBroker`, which [makes the calls without middleware](https://github.com/livewire/livewire/blob/v3.5.8/src/Features/SupportTesting/RequestBroker.php#L28).

Now, the quickest solution is to change `$request->session()->regenerate();` with `session()->regenerate()`, that means that you'll be accessing the session store from the service container instead of from the request, in the case of the tests, the container will regenerate the session from the container even though it's not attached to the current request. In the normal flow of users, the session store returned from the service container will be the same one than the one used from the session because the requests will be coming through the [StartSession middleware](https://github.com/laravel/framework/blob/v11.25.0/src/Illuminate/Session/Middleware/StartSession.php#L51).

I don't have the context to know why the RequestBroker needs to disable all middleware to be tested, that's what it does now.

**Reply by An0nym0usNinja** [reply]
This worked for me thanks!
